sycl/include/sycl/ext/oneapi/matrix/matrix-tensorcore.hpp

```diff
@@ -247,15 +257,27 @@ struct joint_matrix_load_impl<
                              get_layout_id<Layout>());
       }
     } else if constexpr (std::is_same<T, float>::value) {
-      if constexpr (NumRows == 16 && NumCols == 16) {
-        __hmma_m16n16k16_ld_c_f32(res.data, src.get(), stride,
-                                  get_layout_id<Layout>());
-      } else if constexpr (NumRows == 8 && NumCols == 32) {
-        __hmma_m8n32k16_ld_c_f32(res.data, src.get(), stride,
-                                 get_layout_id<Layout>());
-      } else if constexpr (NumRows == 32 && NumCols == 8) {
-        __hmma_m32n8k16_ld_c_f32(res.data, src.get(), stride,
-                                 get_layout_id<Layout>());
+      if (std::is_same<S, float>::value) {
```

Comment:
Is there a reason for this not to be `if constexpr`?

Reply:
I think `if constexpr` is c++17 so it may have been removed for that reason. Although I noticed that for some reason c++17 is allowed in the extension namespace, although I don't understand this?

Reply:
I believe extensions that need to be explicitly included are allowed to use C++17 features.

Reply:
Allowed in the sense that tests don't fail that fail due to c++17 stuff in other namespaces!

Reply:
I see, thanks.

Reply:
> In that case I am happy to ensure c++17 is fully employed where appropriate in this extension in the follow on PR: #5964

As long as it doesn't bleed into sycl.hpp then it should be fine. There should be a test that fails if it does.